### PR TITLE
Align SSL verification options with core for Central commands

### PR DESCRIPTION
**Pull request by lucadruda**
IoT Central commands don't follow CLI core settings for SSL certificate validation

- Added code to handle "AZURE_CLI_DISABLE_CONNECTION_VERIFICATION" environment variable
- Use of "Session" for request module to share options between requests

---
This project has adopted the [Microsoft Open Source Code of Conduct](https://opensource.microsoft.com/codeofconduct/). For more information see the [Code of Conduct FAQ](https://opensource.microsoft.com/codeofconduct/faq/) or contact [opencode@microsoft.com](mailto:opencode@microsoft.com) with any additional questions or comments.

Thank you for contributing to the IoT extension!

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] If introducing new functionality or modified behavior, are they backed by unit and integration tests?
- [x] In the same context as above are command names and their parameter definitions accurate? Do help docs have sufficient content?
- [x] Have **all** unit **and** integration tests passed locally? i.e. `pytest <project root> -vv`
- [x] Have static checks passed using the .pylintrc and .flake8 rules? Look at the CI scripts for example usage.
- [x] Have you made an entry in HISTORY.rst which concisely explains your feature or change?

### CODE REVIEW

File: azext_iot/central/services/api_token.py
Comment (prbans) [inline azext_iot/central/services/api_token.py:13]: nit - remove unused `requests` import

File: azext_iot/central/services/api_token.py
Comment (prbans) [inline azext_iot/central/services/api_token.py:18]: Do we need to make similar changes to how the `monitor-events` work? 

https://github.com/Azure/azure-iot-cli-extension/blob/a5e3accbb036511e618114a988ce19f8af22c9db/azext_iot/common/_azure.py#L122

cc: @digimaun

File: HISTORY.rst
Comment (prbans) [inline HISTORY.rst:6]: I think we typically wait for a full release before we put something in the release history - work with @digimaun on when/how to add release notes

File: azext_iot/central/services/device.py
Comment (prbans) [inline azext_iot/central/services/device.py:24]: Is there a way to write IT's to validate this behavior?

Is there a way to force a change in the session and verify something breaks?

Is there a way to force the SSL to remain the same and see behavior corrected?

File: azext_iot/constants.py
Comment (prbans) [inline azext_iot/constants.py:10]: same version bumping comment - check with @digimaun

File: azext_iot/central/services/device.py
Comment (prbans) [reply azext_iot/central/services/device.py:24]: request changes on review is for this - all else are nits, feel free to #wontfix them (other than if Paymaun has guidance otherwise)

**Review by prbans — CHANGES_REQUESTED** [review]
Great work! Would be good to back this behavior change with IT's

File: HISTORY.rst
Comment (digimaun) [reply HISTORY.rst:6]: This is OK - we now like to update the release history (if there are user facing, or otherwise notable changes) per PR. It helps to more quickly inform the GitHub release notes.

File: HISTORY.rst
Comment (digimaun) [inline HISTORY.rst:9]: I would also include the specific configuration value the user can expect to be aligned i.e. the env var `AZURE_CLI_DISABLE_CONNECTION_VERIFICATION` would be honored.

File: azext_iot/constants.py
Comment (digimaun) [reply azext_iot/constants.py:10]: Yea its a good rule of thumb to check before committing the version in PR. For now we are good with the changes in `0.10.8`.  But keep in mind we could do an out-of-band release (due to the need of an immediate behavior change or fix) before the PR gets merged which would throw this off.

File: azext_iot/central/services/api_token.py
Comment (digimaun) [reply azext_iot/central/services/api_token.py:18]: I think we should be consistent and that function should be at play due to the `requests.post` against a central app endpoint. Good catch.

File: HISTORY.rst
Comment (lucadruda) [reply HISTORY.rst]: added

File: azext_iot/common/_azure.py
Comment (prbans) [inline azext_iot/common/_azure.py:123]: how come we arent using `http` here?

Comment (prbans) [inline azext_iot/common/_azure.py:123]: @digimaun - does this need to be done for the hub monitor events too?

**Review by lucadruda — COMMENTED** [review]
Added comments

File: azext_iot/common/_azure.py
Comment (lucadruda) [reply azext_iot/common/_azure.py:123]: "http" is just a variable name. Also there is no reason to use Session here as there's just one call and verification settings can be put in the post options. Session is useful in other files where multiple requests need to share same settings

**Comment by lucadruda** [comment]
Change session back to single execution. The verify option is passed to every requests.
This helps when testing fine grained.
Converting to draft until tests ready.

File: azext_iot/common/_azure.py
Comment (digimaun) [reply azext_iot/common/_azure.py:123]: I am not aware of a requirement for this functionality in hub monitor-events. So I'm leaning towards no in this PR. We also need to verify `uamqp` supports the functionality of a TLS connection without verification.

**Comment by digimaun** [comment]
Closing due to inactivity.

**Pull request closed without merging**
